lms/djangoapps/progress/signals.py

```diff
@@ -26,16 +27,24 @@
 log = logging.getLogger(__name__)
 
 
-def _get_parent_content_id(html_content_id):
-    """ Gets parent block content id """
+def is_valid_progress_module(content_id):
+    """
+    Returns boolean indicating if given module is valid for marking progress
+    A valid module should be child of `vertical` and its category should be
+    one of the PROGRESS_DETACHED_CATEGORIES
+    """
     try:
-        html_usage_id = BlockUsageLocator.from_string(html_content_id)
-        html_module = modulestore().get_item(html_usage_id)
-        return unicode(html_module.parent)
+        detached_categories = getattr(settings, 'PROGRESS_DETACHED_CATEGORIES', [])
+        usage_id = BlockUsageLocator.from_string(content_id)
+        module = modulestore().get_item(usage_id)
+        return module and \
+            module.parent and \
+            module.parent.category == "vertical" \
+            and module.category not in detached_categories
     except (InvalidKeyError, ItemNotFoundError) as exception:
         # something has gone wrong - the best we can do is to return original content id
```

Comment:
We are no longer returning the 'original content id' here, we are now returning False -- switching from a string response to a boolean is a semantic change for this method, so we'll need to ensure we've updated any clients which might be expecting a content_id value.